library(DESeq2)
library(edgeR)
library(org.Hs.eg.db)
library(pheatmap)
library(viridis)
library(AnnotationDbi)

## Prepare count matrix from bam files with the following command
# featureCounts -p -g gene_id -a gencode.v37.annotation.gtf -T 12 -o count_matrix.txt *bam

##Differential gene expression##
#Import count table generated by featureCounts and make metadata
a<-read.table("MYCi_count_matrix.txt",h=TRUE)
a2<-a[,c(1,7:ncol(a))] #Removes chr, start, end for exons which take up a lot of space

# Make metadata
grp1 <- c("D484_S68_.bam","D485_S69_.bam","D486_S70_.bam")
grp2 <- c("D487_S71_.bam","D488_S72_.bam","D489_S73_.bam")
grp3 <- c("D490_S74_.bam","D491_S75_.bam","D492_S76_.bam")
cond1="Crtl"
cond2="MYCi6"
cond3="MYCi975"

condition=factor(c(rep(cond1,length(grp1)),rep(cond2,length(grp2)),rep(cond3,length(grp3))))
condition <- relevel(condition, ref=cond1)
replicate<-factor(c("a","b","c","a","b","c","a","b","c"))

# Mate design tabel for potential batch normalization
sampleTable <- data.frame(
  row.names=c(grp1,grp2,grp3),
  replicate=replicate,
  condition=condition)
sampleTable

n=2 #Set number of comparisons. This will be used for deseq2. The list below determines the actual comparisons. 
conList<-list(c("condition","MYCi6","Crtl"),
              c("condition","MYCi975","Crtl"))

# Annotate with gene names
gene <- a$Geneid
gene_info <- strsplit(gene, "\\.")
gene_names <- lapply(gene_info, function(x){x[1]})
gene_names <- as.data.frame(unlist(gene_names))
colnames(gene_names) <- c("ensembl_1")
df <- cbind(a, gene_names) 
Genes <- as.vector(df$ensembl_1)
Genes <- (mapIds(org.Hs.eg.db, Genes, 'SYMBOL',"ENSEMBL"))
a3 <- a2
a3$symbol <- Genes


#Attach TSS coordinates to genes using the longest version of the gene.
a4<-a3[rowSums(a3[,2:ncol(a2)])>=20,] #Make count table ready for deseq2. Include only expressed genes with >=10 tags i all conditions.

#Make differential analysis using DESeq2
library(DESeq2)
a5 <- DESeqDataSetFromMatrix(countData = a4[,2:ncol(a2)], colData = sampleTable, design = ~condition)
a5 <- DESeq(a5) #Run diffexp analysis


#Attach padj and logFC for each comparison to raw data
a6 <- gene_count <- as.data.frame(counts(a5, normalized=TRUE))
a6$ensembl <- a4$Geneid
a6$symbol <- a4$symbol
diff<-a6 # Normalized counts
for (i in 1:n) {
  res<-data.frame(results(a5,contrast = conList[[i]])) #Get p values for different comparisons 
  colnames(res)[c(2,6)]<-c(paste("log2FC",paste(".c",i,sep = ""),sep = ""),paste("padj",paste(".c",i,sep = ""),sep = ""))
  diff<-cbind(diff,res[,c(2,6)])
}


#Make PCA plot fo subset of samples without the 468 cell line, which dominates the PCA
rld<-rlog(a5,blind = F)
plotPCA(rld) 

## after running example("plotPCA")
z <- plotPCA(rld)
z + geom_label(aes(label = name))



enh <- read.delim("NewInh_enh_down.txt")
enh$Ann <- sub(" .*","",enh$Annotation)
enh <- enh[enh$Ann == "Intergenic",]

enh <- enh[(enh$Veh_merged.bam.TD.Tag.Count.in.2000.bp..21482277.5.Total..normalization.factor...0.47..effective.total...10000000. + enh$MYCi975_merged.bam.TD.Tag.Count.in.2000.bp..23446837.5.Total..normalization.factor...0.43..effective.total...10000000. + enh$MYCi6_merged.bam.TD.Tag.Count.in.2000.bp..19137658.5.Total..normalization.factor...0.52..effective.total...10000000.) > 2,]


boxplot(log2(enh$MYCi975_merged.bam.TD.Tag.Count.in.2000.bp..23446837.5.Total..normalization.factor...0.43..effective.total...10000000./enh$Veh_merged.bam.TD.Tag.Count.in.2000.bp..21482277.5.Total..normalization.factor...0.47..effective.total...10000000.),
        log2(enh$MYCi6_merged.bam.TD.Tag.Count.in.2000.bp..19137658.5.Total..normalization.factor...0.52..effective.total...10000000./enh$Veh_merged.bam.TD.Tag.Count.in.2000.bp..21482277.5.Total..normalization.factor...0.47..effective.total...10000000.), outline = F)


t.test(enh$Veh_merged.bam.TD.Tag.Count.in.2000.bp..21482277.5.Total..normalization.factor...0.47..effective.total...10000000.,
       enh$MYCi975_merged.bam.TD.Tag.Count.in.2000.bp..23446837.5.Total..normalization.factor...0.43..effective.total...10000000., paired = T)

t.test(enh$Veh_merged.bam.TD.Tag.Count.in.2000.bp..21482277.5.Total..normalization.factor...0.47..effective.total...10000000.,
       enh$MYCi6_merged.bam.TD.Tag.Count.in.2000.bp..19137658.5.Total..normalization.factor...0.52..effective.total...10000000., paired = T)


Enhancer <- read.delim("NewInh_subenh_down_agg.txt")
avr1_enh <-Enhancer[,c(1,grep("Coverage", colnames(Enhancer)))]
colnames(avr1_enh)<-c("dist","MYCi6","MYCi975","Veh")

# KDM3A upon KJ in MYC activated enhancer
plot <-melt(avr1_enh, id="dist")

ggplot(plot, aes(x=dist,y=value, col=variable))+
  geom_line(size=2)+
  scale_fill_viridis_d(option="E") + scale_color_viridis_d(option="E") +
  theme(axis.title = element_text(size = 20),
        axis.text = element_text(size = 20, colour = "black"),
        panel.background = element_blank(),
        panel.grid = element_blank(),
        axis.line = element_line(color = "black"),panel.border = element_rect(colour = "black", fill = NA), legend.position = "bottom")+
  geom_vline(xintercept = 0, linetype="dashed") + ylab("TagCounts") + xlab("bp") 



# Save dataframe
diff <- read.table("DE_analysis_NewInhib_BT549.txt")
diff <- diff[complete.cases(diff),]
KO <- read.table("DE_analysis_KO_BT549.txt")
KO <- KO[complete.cases(KO),]
KJ <- read.table("DE_analysis_KJ_BT549_DEX_new.txt")
KJ <- KJ[complete.cases(KJ),]

merged <- merge(KJ, diff, by ="symbol")
merged <- merge(merged, KO, by ="symbol")

plot <- merged[merged$padj.c1 < 0.05 | merged$padj.c2 < 0.05,]

ggplot(merged, aes(x=(plot$log2FC.c2), y=(plot$log2FC.c1))) +
  geom_point(alpha = 0.2) +geom_hline(yintercept=0) + geom_smooth(method = "lm", col = "black", se = F, formula=y~x-1) + stat_cor(method = "spearman", size = 5) + 
  geom_vline(xintercept=0) + scale_fill_viridis_c(option = "E") +
  theme(axis.title = element_text(size = 20),
        axis.text = element_text(size = 20, colour = "black"),
        panel.background = element_blank(),
        panel.grid = element_blank(),
        axis.line = element_line(color = "black"),panel.border = element_rect(colour = "black", fill = NA)) 


KJ_down <- merged[(merged$padj.c1 < 0.05 & merged$log2FC.c1 < -.0) & (merged$padj.c2 < 0.05 & merged$log2FC.c2 < -.0)  , ]
KJ_up  <- merged[(merged$padj.c1 < 0.05 & merged$log2FC.c1 > .50) & (merged$padj.c2 < 0.05 & merged$log2FC.c2 > .50), ]


boxplot(KJ_down$log2FC.c1,
        KJ_up$log2FC.c1,
        KJ_down$log2FC.c2,
        KJ_up$log2FC.c2,
        KJ_down$Log2FC_MYC_vs_Crtl.x,
        KJ_up$Log2FC_MYC_vs_Crtl.x,
        outline = F)




BT549_count <- read.delim("MYC_BT549_Consensus.count.txt")

# Identify MYC promoters and enhancers
BT549_count$Ann <- sub(" .*","",BT549_count$Annotation)
BT549_anno_Enhancer <- BT549_count[BT549_count$Ann == "Intergenic" | BT549_count$Ann == "intron",]
BT549_anno_Enhancer <- BT549_anno_Enhancer[(BT549_anno_Enhancer$Distance.to.TSS < 50000 & BT549_anno_Enhancer$Detailed.Annotation > 3000) & BT549_anno_Enhancer$Distance.to.TSS > -50000,]
BT549_anno_Promoter <- BT549_count[BT549_count$Ann == "promoter-TSS",]
BT549_anno_Promoter <- BT549_anno_Promoter[!duplicated(BT549_anno_Promoter$Gene.Name),]

# Summarise enhancer contribution 
BT549_anno_Enhancer_strengh <- BT549_anno_Enhancer[,c("Gene.Name","...MYC_Project.MYC_MED1_ChIP.tagdir.Combined_MYC_BT549..Tag.Count.in.1000.bp..40939315.0.Total..normalization.factor...0.24..effective.total...10000000.")]
BT549_anno_Enhancer_strengh %>% group_by(Gene.Name) %>% summarise_all(funs(sum)) -> BT549_anno_Enhancer_strengh
colnames(BT549_anno_Enhancer_strengh) <- c("symbol","Count_enh")

# Summarise promoter contribution 
BT549_anno_promoter_strengh <- BT549_anno_Promoter[,c("Gene.Name","...MYC_Project.MYC_MED1_ChIP.tagdir.Combined_MYC_BT549..Tag.Count.in.1000.bp..40939315.0.Total..normalization.factor...0.24..effective.total...10000000.")]
BT549_anno_promoter_strengh %>% group_by(Gene.Name) %>% summarise_all(funs(sum)) -> BT549_anno_promoter_strengh
colnames(BT549_anno_promoter_strengh) <- c("symbol","Count_prom")

# Merge genes with either promoter or enhancer MYC contribution
plot <- full_join(BT549_anno_Enhancer_strengh, BT549_anno_promoter_strengh)
plot[is.na(plot)] = 0

# Calculate the difference en contribution from different regions
plot$count <- plot$Count_prom-plot$Count_enh

# Add gene expression information to the genes with nearby MYC binding if the gene is significant regulated upon siRNA knockdown (data from DESeq2 script)
plot2 <- read.table("DE_analysis_KO_BT549.txt")
plot2 <- plot2[plot2$padj_MYC_vs_Crtl < 1,]
plot2 <- plot2[,c("symbol", "Log2FC_MYC_vs_Crtl")]
plot3 <- left_join(plot, plot2)
plot3[is.na(plot3)] = 0
plot3 <- plot3[!duplicated(plot3$symbol),]

# Only include expressed genes
plot3 <- plot3[abs(plot3$Log2FC_MYC_vs_Crtl) > 0,]

# Make GCN5 geneset
# All significant regulated
cluster1_BT <- as.data.frame(diff[diff$padj.c1 < 0.05  ,]$symbol)
colnames(cluster1_BT) <- c("symbol")
cluster1_BT$geneset <- rep("MB3_sig")
cluster1_BT <- cluster1_BT[,c("geneset","symbol")]
clust1_tibble <- as.tibble(cluster1_BT)

# All non significant regulated
cluster2_BT <- as.data.frame(diff[diff$padj.c1 > 0.05,]$symbol)
colnames(cluster2_BT) <- c("symbol")
cluster2_BT$geneset <- rep("MB3_not")
cluster2_BT <- cluster2_BT[,c("geneset","symbol")]
clust2_tibble <- as.tibble(cluster2_BT)

# All significant down regulated
cluster3_BT <- as.data.frame(diff[diff$padj.c1 < 0.05 & diff$log2FC.c1 < 0,]$symbol)
colnames(cluster3_BT) <- c("symbol")
cluster3_BT$geneset <- rep("MB3_down")
cluster3_BT <- cluster3_BT[,c("geneset","symbol")]
clust3_tibble <- as.tibble(cluster3_BT)

# All significant up regulated
cluster4_BT <- as.data.frame(diff[diff$padj.c1 < 0.05 & diff$log2FC.c1 > 0,]$symbol)
colnames(cluster4_BT) <- c("symbol")
cluster4_BT$geneset <- rep("MB3_up")
cluster4_BT <- cluster4_BT[,c("geneset","symbol")]
clust4_tibble <- as.tibble(cluster4_BT)

# Create list of genesets for enrichment analysis
clusters <- rbind(clust1_tibble,clust2_tibble,clust3_tibble,clust4_tibble)
clusters$symbol <- as.integer(mapIds(org.Hs.eg.db, clusters$symbol,"ENTREZID", "SYMBOL"))
colnames(clusters) <- c("gs_name","entrez_gene")

# Bin gene based on MYC promoter and enhancer binding
library(mltools)
plot3[, "group"] <- bin_data(plot3$count, bins=4, binType = "quantile")

# Subset bins - bin 1 is promoter dominated and bin 4 is most enhancer domincated
bin1 <- plot3[plot3$group =="[71.57, 700.43]",]
bin2 <- plot3[plot3$group =="[25.53, 71.57)",]
bin3 <- plot3[plot3$group =="[-48.73, 25.53)",]
bin4 <- plot3[plot3$group =="[-1079.65, -48.73)",]

# Calculate enrichment for each gene bin
library(clusterProfiler)
Genes <- (mapIds(org.Hs.eg.db, bin1$symbol,"ENTREZID", "SYMBOL"))
bin1 <- enricher(Genes, TERM2GENE=clusters, pvalueCutoff =1, maxGSSize = 30000)
bin1 <- as.data.frame(bin1@result)
bin1$type <- rep("bin1")

Genes <- (mapIds(org.Hs.eg.db, bin2$symbol,"ENTREZID", "SYMBOL"))
bin2 <- enricher(Genes, TERM2GENE=clusters, pvalueCutoff =1, maxGSSize = 30000)
bin2 <- as.data.frame(bin2@result)
bin2$type <- rep("bin2")

Genes <- (mapIds(org.Hs.eg.db, bin3$symbol,"ENTREZID", "SYMBOL"))
bin3 <- enricher(Genes, TERM2GENE=clusters, pvalueCutoff =1, maxGSSize = 30000)
bin3 <- as.data.frame(bin3@result)
bin3$type <- rep("bin3")

Genes <- (mapIds(org.Hs.eg.db, bin4$symbol,"ENTREZID", "SYMBOL"))
bin4 <- enricher(Genes, TERM2GENE=clusters, pvalueCutoff =1, maxGSSize = 30000)
bin4 <- as.data.frame(bin4@result)
bin4$type <- rep("bin4")

# Merge for each bin
plot <- rbind(bin1,bin2,bin3,bin4)

# Subset GCN5 down-regulated genes and plot
plot_sig <- plot[plot$ID == "MB3_sig" | plot$ID == "MB3_not",]

library(ggplot2)
ggplot(data=plot_sig, aes(y=-log10(p.adjust), x=ID, fill = type)) + geom_bar(stat="identity", position = "dodge") + 
  theme(axis.title = element_text(size = 20),
        axis.text = element_text(size = 20, colour = "black"),
        panel.background = element_blank(),
        panel.grid = element_blank(),
        axis.line = element_line(color = "black"),panel.border = element_rect(colour = "black", fill = NA), legend.position = "bottom")  + xlab(" ") + ylab("-log10(q-value)")






library(fgsea)

gseaDat <- filter(diff, !is.na(diff$symbol))
ranks <- gseaDat$log2FC.c2
names(ranks) <- gseaDat$symbol
head(ranks)

pathways <- gmtPathways("msigdb.v7.4.symbols.gmt")

fgseaRes <- fgsea(pathways, ranks, minSize=15, maxSize = 10000, nperm=1000)

head(fgseaRes[order(-NES, abs(NES)), ], n=20)


plotEnrichment(pathways[["WHITFIELD_CELL_CYCLE_S"]], ranks) + 
  theme(axis.title = element_text(size = 20),
        axis.text = element_text(size = 20, colour = "black"),
        panel.background = element_blank(),
        panel.grid = element_blank(),
        axis.line = element_line(color = "black"),panel.border = element_rect(colour = "black", fill = NA), legend.position = "bottom")+
  theme(axis.text.x=element_blank(),
        axis.ticks.x=element_blank()) + ylab("Enrichment Score") + xlab("Ranked MYC Promter - Distal")



